### ISSUE TITLE
Nginx-proxy manager add ability to run as a different user

### ISSUE THREAD

**Issue by ejalal** [truenas/charts#2338]
Hello

I'm new to TrueNAS Scale (used OMV for 5 years now) and I'm trying to install nginx proxy manager from official charts (not truecharts).

Having issues with app folder permissions as I would like to run all apps with the "apps" user and group (created a dedicated apps dataset belongs to apps:apps to allow easy backup/restore)

The thing with nginx-proxy-manager does not allow to setup the app to run with a different user, and the default one set in the charts is 1000:1000 which doesn't exist on TrueNAS Scale 23.10 creates a few permission errors while starting the app.

If would be interesting to be able to setup the app to run as 568:568 from the install wizard like most of the other apps (jellyfin, radarr...)

Thanks

**Comment by stavros-k** [comment]
Hello, ideally I would like all apps to be able to run with any uid the user selects. Both from security perspective and usability.
Unfortunately this is not always possible, for example NPM runs as root always, due to upstream restrictions.
I saw a recent change that it might allow us to run the **sub**-processes as a different UID (currently defaults to 1000 specified by upstream).
But the parent process will be root for that application. 

TL;DR if the container can run as a non-root user and can be user-defined, I make sure to expose that setting!

**Comment by ejalal** [comment]
Thanks for the quick responde @stavros-k, much appreciated.

I'm more familiar with docker compose and docker run and I'm pretty sure I was running all my containers as non root by pushing PUID and PGID environment variables to the container.

Tried to do the same here as a workaround but was rejected with this error "Environment Variable [PGID] in [envList] tried to override the Environment Variable that is already defined in [fixedEnv]"
